### DISCUSSION TITLE
Range in notification templates?

### DISCUSSION THREAD

**Discussion by bianchi2** [argoproj/argo-cd#15145] [Q&A]
I have multiple sources in my application spec. Is it possible to range through them or access by number? `.app.spec.sources[1].repoUrl` is unacceptable in a template and notification service complains. I tried using range, but that does not seem to be supported.

**Comment by qxmips** [comment]
is there a solution for that?

**Reply by qxmips** [reply]
this did the trick for me:
```
   {{if .app.spec.source.repoURL}}
          {
            "title": "Repository",
            "value": "{{.app.spec.source.repoURL}}",
            "short": true
          }
  {{else}}
          {
              "title": "Repository",
              "value": "{{ (index .app.spec.sources 0).repoURL }}",
              "short": true
          }
   {{end}}
````
